salt/orchestrate/aws/s3_buckets/edx.sls

```diff
@@ -0,0 +1,50 @@
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% for env in ['mitxpro-qa', 'mitxpro-production', 'mitx-qa', 'mitx-production'] %}
+{% set env_data = env_settings.environments[env] %}
+{% for purpose in env_data.purposes %}
+{% set bucket_prefix = env_data.secret_backends.aws.bucket_prefix %}
+
+video_uploads_bucket_{{ purpose }}_{{ env }}:
+  boto_s3_bucket.present:
+    - Bucket: {{ bucket_prefix }}-edx-video-upload-{{ purpose }}-{{ env }}
+    - Versioning:
+        Status: Enabled
+    - region: us-east-1
+    - Tagging:
+        OU: mitxpro
+        business_unit: mitxpro
+        Department: mitxpro
+        Environment: {{ env }}
+    - ACL:
+        GrantRead: "uri=http://acs.amazonaws.com/groups/global/AllUsers"
+    - Policy:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: "PublicRead"
+            Effect: "Allow"
+            Principal: "*"
+            Action: "s3:GetObject"
+            Resource: "arn:aws:s3:::{{ bucket_prefix }}-edx-video-upload-{{ purpose }}-{{ env }}/*"
+
+{% for use in bucket_uses %}
```

Comment:
This loop is going to override the values set in the prior step for the video uploads bucket, the same as what we ran into previously.

We can either flip the ordering, add a conditional to exclude uploads from this loop, or add a conditional to specify the policy within this loop rather than having it as a separate step.

Reply:
That loop specifies a different bucket name though - `xpro-app-{{ env }}`